vos/vos/vos.py

```diff
@@ -2732,6 +2762,78 @@ def delete(self, uri):
+            response.raise_for_status()
+            if response.status_code != 303:
+                raise RuntimeError('Unexpected response for running job: '
+                                   + response.status_code)
+            return self._run_recursive_job(session, response.headers['location'])
+
+    def _run_recursive_job(self, session, url):
+        # runs an already created recursive job and returns the number of
+        # successfull and unsuccessfull actions
+        logger.debug('POST: ' + url)
+        response = session.post(url + '/phase', data={'phase': 'RUN'}, allow_redirects=False)
+        if response.status_code != 303:
+            raise RuntimeError('Unexpected response for running job: '
+                               + response.status_code)
+
+        # polling: WAIT will block for up to 6 sec or until phase change or if job is in
+        # a terminal phase
+        jobPoll = url + '?WAIT=6'
+        count = 0
+        done = False
+        while not done and count < 100:  # max 100*6 = 600 sec polling
+            logger.debug("poll: " + jobPoll)
+            resp = session.get(jobPoll)
+            resp.raise_for_status()
+            xml_string = resp.content
+            logging.debug('Job Document:{}'.format(xml_string))
+            job_document = ElementTree.fromstring(xml_string)
+            if job_document.find('uws:phase', UWS_NSMAP) is not None:
+                phase = job_document.find('uws:phase', UWS_NSMAP).text
+            else:
+                raise RuntimeError('Cannot determine job phase')
+            if phase.upper() in ['QUEUED', 'EXECUTING']:
+                count += 1
+            elif phase.upper() == 'ERROR':
+                message = 'Failed'
+                error_summary = job_document.find('uws:errorSummary', UWS_NSMAP)
+                if (error_summary is not None and
+                        error_summary.find('uws:message', UWS_NSMAP) is not None):
+                    message = error_summary.find('uws:message', UWS_NSMAP).text
+                raise RuntimeError(message)
+            elif phase.upper() in ['COMPLETED', 'ABORTED']:
+                results = job_document.find('uws:results', UWS_NSMAP)
+                error_count = 0
+                success_count = 0
+                if results is not None:
+                    results = results.findall('uws:result', UWS_NSMAP)
+                    if results is not None:
+                        for res in results:
+                            if res.attrib['id'] == 'successcount':
+                                success_count = res.attrib['{' + UWS_NSMAP['xlink'] + '}href'].split(':')[1]
+                            elif res.attrib['id'] == 'errorcount':
+                                error_count = res.attrib['{' + UWS_NSMAP['xlink'] + '}href'].split(':')[1]
+                return success_count, error_count
+            else:
+                raise RuntimeError('Unknow job phase: ' + phase)
```

Comment:
Will this function encounter either the PENDING or SUSPENDED states? Are those both considered `RuntimeError`s?

Reply:
In theory, it can go into the `SUSPENDED` state although I don't think that we have that in practice. I've added to the states anyhow. It is a mechanism for the service providers to signal a backend resource problems but the job moves automatically back into `EXECUTING` after that is addressed so not sure what can the client do differently than when it's in the `EXECUTING` phase. 
There's no `PENDING` once the job is started when the client sends the `RUN` command.